src/app/public/modules/list-view-grid/list-view-grid.component.ts

```diff
@@ -231,12 +242,18 @@ export class SkyListViewGridComponent
     this.gridState.map(s => s.columns.items)
       .take(1)
       .subscribe(columns => {
-        let displayedColumns = selectedColumnIds.map(
-          columnId => columns.filter(c => c.id === columnId)[0]
-        );
-        this.gridDispatcher.next(
-          new ListViewDisplayedGridColumnsLoadAction(displayedColumns, true)
-        );
+        this.selectedColumnIds
```

Comment:
Just thinking out loud here... Is all this comparison logic too complicated for a `distinctUntilChanged()`? I see that you're fetching column Ids from an observable so it certainly is a bit more complex than a normal comparison. Just curious if you can think of a way to make this comparison check more consistent with the rest of our observables...

Reply:
The key here is that we aren't comparing the old version of the observable to the new version we are comparing the new version of the observable to a local variable. 

However, looking at it again I do think that I can switch the order of the observables here so that if the if statement fails we don't even grab the value of the grid state.

Reply:
Updated.